src/box/lua/schema.lua

```diff
@@ -3718,13 +3769,26 @@ box.schema.user.create = atomic_wrapper(function(name, opts)
                    nil, {if_not_exists=true})
 end)
 
-box.schema.user.exists = function(name)
+box.schema.user.exists = function(name, opts)
     utils.box_check_configured(2)
-    if user_resolve(name, 2) then
+    local origin
+    if opts ~= nil then
+        check_param_table(opts, { _origin = 'string' }, 2)
+        origin = opts._origin
+    end
+    local uid = user_resolve(name, 2)
+    if uid == nil then
+        return false
+    end
+    if origin == nil then
         return true
-    else
+    end
```

Comment:
To think: is it OK that `opts = nil` and `opts = {_origin = ''}` works differently here? It works the same for all the other functions that accept `opts = {_origin = <...>}`.

I guess that we really need that `box.schema.{user,role}.exists()` checks a real presence of a user, not just its presence in the default origin, because it may be used in some privilege check logic, for example.

Need to spin around those thoughs for a while.